### PR TITLE
Skip module access check for processMethod()

### LINKED ISSUE #11274: JPMS IllegalAccessError
Java -version output
--------------------
openjdk version "15.0.1" 2020-10-20
OpenJDK Runtime Environment AdoptOpenJDK (build 15.0.1+9)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.23.0, JRE 15 Linux amd64-64-Bit Compressed References 20201022_81 (JIT enabled, AOT enabled)
OpenJ9   - 0394ef754
OMR      - 582366ae5
JCL      - ad583de3b5 based on jdk-15.0.1+9)

openjdk version "15.0.1" 2020-10-20
OpenJDK Runtime Environment AdoptOpenJDK (build 15.0.1+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 15.0.1+9, mixed mode, sharing)

Summary of problem
------------------
Running a JPMS application with OpenJ9 throws IllegalAccessError while with Hotspot it does not.

Diagnostic files
----------------
[bug-jpms.zip](https://github.com/eclipse/openj9/files/5598071/bug-jpms.zip)

```
$ mvn clean package
$ java --module-path target/bug-jpms-1.0-SNAPSHOT.jar:target/dependency -m jpms.example/jpms.example.Main
```

Output OpenJ9
```
Exception in thread "DefaultHttpServer-ProcessingThread-0" java.lang.IllegalAccessError: Class cloud/piranha/http/nano/NanoHttpServerProcessor(module cloud.piranha.http.nano) can not access class cloud/piranha/http/webapp/HttpWebApplicationResponse(module cloud.piranha.http.webapp) because module module cloud.piranha.http.nano does not read module module cloud.piranha.http.webapp 
        at cloud.piranha.http.nano/cloud.piranha.http.nano.NanoHttpServerProcessor.process(NanoHttpServerProcessor.java:66)
        at cloud.piranha.http.impl/cloud.piranha.http.impl.DefaultHttpServerProcessingThread.run(DefaultHttpServerProcessingThread.java:85)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at java.base/java.lang.Thread.run(Thread.java:845)
```

### PR DESCRIPTION
**Pull request by JasonFengJ9**
The module access check has been performed at `class/interface` resolving time, can be skipped within `processMethod()`.

fixes #11274 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by JasonFengJ9** [comment]
@tajila could you please review?

File: runtime/vm/visible.c [hunk 1]
```diff
@@ -85,6 +85,12 @@ checkModuleAccess(J9VMThread *currentThread, J9JavaVM* vm, J9ROMClass* srcRomCla
  */
 IDATA
 checkVisibility(J9VMThread *currentThread, J9Class* sourceClass, J9Class* destClass, UDATA modifiers, UDATA lookupOptions)
+{
+	return checkVisibilityModuleAccessOption(currentThread, sourceClass, destClass, modifiers, lookupOptions, TRUE);
```
Comment (tajila) [inline runtime/vm/visible.c:89]: instead of this can we just introduce a `J9_LOOK_NO_MODULARITY_FLAG` ?

File: runtime/vm/visible.c [hunk 2]
```diff
@@ -100,7 +106,8 @@ checkVisibility(J9VMThread *currentThread, J9Class* sourceClass, J9Class* destCl
 	if (!J9CLASS_IS_EXEMPT_FROM_VALIDATION(sourceClass)) {
 		if ( modifiers & J9AccPublic ) {
 			/* Public */
-			if ((sourceClass != destClass)
+			if (doModuleAccessCheck
```
Comment (tajila) [inline runtime/vm/visible.c:109]: The flag can be used here

File: runtime/vm/visible.c (hunk 1 above)
Comment (tajila) [reply runtime/vm/visible.c:89]: i just changed the name, last one didnt really make sense

Comment (JasonFengJ9) [reply runtime/vm/visible.c]: Good idea, use `J9_LOOK_NO_MODULE_CHECKS` instead. Please have another look.

File: runtime/vm/visible.c (hunk 2 above)
Comment (JasonFengJ9) [reply runtime/vm/visible.c]: yeah, check via `J9_ARE_NO_BITS_SET(lookupOptions, J9_LOOK_NO_MODULE_CHECKS)`.

**Comment by tajila** [comment]
Jenkins test sanity win jdk8

**Comment by tajila** [comment]
Jenkins test sanity,extended plinux jdk11

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
@JasonFengJ9 this needs to be double delivered in the 0.24 stream

**Comment by tajila** [comment]
Or maybe check with @pshipton, perhaps he can just push it.

**Comment by JasonFengJ9** [comment]
Checked with @pshipton , double delivery is required, just created https://github.com/eclipse/openj9/pull/11413.

**Pull request merged**